### ISSUE TITLE
[AospMusic][v1.3]Can not add song to playlist.

### ISSUE THREAD

**Issue by masudr4n4** [hvdwolf/aosp-music#6]
<h4>Actual behavior</h4>
The app crashed when trying to add a song to a play list.
<h4>Expected behavior</h4>
I expected that I can add a song to any playlist succesfully without crashing the app.
<h4>How to reproduce</h4>
<ol>
<li> Open the app and give all the permission</li>
<li>Click "Songs"</li>
<li>Choose a song ,click and hold to the song </li>
<li>Option will appear then click add to playlist</li>
<li>Choose any playlist to add the song</li>
<li>See the error</li>
</ol>
<h4>Information</h4>
<ul>
<li>App version:v1.3</li>
<li>Os:Android9.0</li>
</ul>
<h4>Recording of the bug</h4>
<a href="https://youtu.be/NdguQg57Ncw">https://youtu.be/NdguQg57Ncw</a>
<h4>Logcat</h4>
<pre><code>
04-30 19:51:38.545 13011 13011 E AndroidRuntime: FATAL EXCEPTION: main
04-30 19:51:38.545 13011 13011 E AndroidRuntime: Process: com.android.music, PID: 13011
04-30 19:51:38.545 13011 13011 E AndroidRuntime: java.lang.SecurityException: Permission Denial: writing com.android.providers.media.MediaProvider uri content://media/external/audio/playlists/100/members from pid=13011, uid=10193 requires android.permission.WRITE_EXTERNAL_STORAGE, or grantUriPermission()
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:1966)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:1934)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.content.ContentProviderProxy.bulkInsert(ContentProviderNative.java:498)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.content.ContentResolver.bulkInsert(ContentResolver.java:1659)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at com.android.music.MusicUtils.addToPlaylist(MusicUtils.java:583)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at com.android.music.TrackBrowserActivity.onContextItemSelected(TrackBrowserActivity.java:676)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.app.Activity.onMenuItemSelected(Activity.java:3650)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at com.android.internal.policy.PhoneWindow$PhoneWindowMenuCallback.onMenuItemSelected(PhoneWindow.java:3948)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at com.android.internal.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:776)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at com.android.internal.view.menu.SubMenuBuilder.dispatchMenuItemSelected(SubMenuBuilder.java:82)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at com.android.internal.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:155)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:923)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:913)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at com.android.internal.view.menu.MenuPopup.onItemClick(MenuPopup.java:128)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.widget.AdapterView.performItemClick(AdapterView.java:362)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.widget.AbsListView.performItemClick(AbsListView.java:1689)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.widget.AbsListView$PerformClick.run(AbsListView.java:4130)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.widget.AbsListView$7.run(AbsListView.java:6612)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:873)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7073)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-30 19:51:38.545 13011 13011 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
</code>
</pre>

**Comment by maaudrana** [comment]
Hi @hvdwolf  can you pls chack those issue i created?

**Comment by maaudrana** [comment]
Hi @hvdwolf you there?

**Comment by hvdwolf** [comment]
I have been very busy the last 2-3 weeks. I did however look into the issues but I have no idea how to solve them. I can't find any related "bug fixes" in the original code either.
I did not create this music player. This music player is only a  "visually changed" standard AOSP music player. I did not even touch the code.
I created this "big button" version as I have an Android car head unit running at 1024x600. Being in the car I need big buttons.
The only "addition" is some intents in the AndroidManifest.xml to add some specific intents used on my Joying head unit.

So sorry to say, but most probably all the mentioned bugs are not going to be solved. I simply do not have the skills. (It would also require to fix the original code.)
